config/config.go

```diff
@@ -45,8 +46,8 @@ func (c *Config) GetTargetFor(path string) *Target {
 
 var GlobalCfg *Config
 
-func Read(targetProfile string, upstreamProfile string, threads int, strExecutor func(s string) (string, error)) (*Config, error) {
-	project, err := readDBTProject()
+func Read(targetProfile string, upstreamProfile string, threads int, customConfigPath string, strExecutor func(s string) (string, error)) (*Config, error) {
```

Comment:
I assume the Read function is not used anywhere else?

Reply:
I'm surprized that no tests are failing... Are these code paths not covered by unit tests ? If so, should we add some ?

Reply:
Refactored the config logic out into a separate function and added tests for it